src/EntityFramework/Core/EntityClient/DbParameterCollectionHelper.cs

```diff
@@ -292,6 +292,50 @@ private static int IndexOf(IEnumerable items, string parameterName)
             return -1;
         }
 
+        /// <inhertidoc />
+        [SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1604:ElementDocumentationMustHaveSummary")]
+        [SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1611:ElementParametersMustBeDocumented")]
+        [SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1615:ElementReturnValueMustBeDocumented")]
+        [SuppressMessage("Microsoft.Usage", "CA2201:DoNotRaiseReservedExceptionTypes")]
+        public DbParameter GetParameterByValue(string value)
+        {
+            var index = IndexOfParameterValue(InnerList, value);
+            if (index < 0)
+            {
+                return null;
+            }
+
+            return InnerList[index];
+        }
+
+        private static int IndexOfParameterValue(IEnumerable items, string parameterValue)
```

Comment:
It's down to globalisation.
The first comparison does a standard does stringA match stringB, nice efficient and quick.  The second does a match based on the Culture which is slower, and will be needed by far fewer clients. 
My understanding is that in the majority of cases the first condition is adequate. I think the reason for the second check was mainly around the StringCompareOption of IgnoreKanaType.  Rather than me attempting to explain, best bet is to view the MSDN article on it - https://msdn.microsoft.com/en-us/library/system.data.sqltypes.sqlstring.ignorekanatype(v=vs.110).aspx